### PR TITLE
Copy over the default tax category to line items if the variant does not have one specified.

### PR DESCRIPTION
**Pull request by athal7**
This seems like the behavior that is expected for 'default', it is weird
for a line item not to have a tax category.

### CODE REVIEW

**Comment by gmacdougall** [comment]
:+1:

File: core/app/models/spree/line_item.rb [hunk 1]
```diff
@@ -47,9 +47,7 @@ def copy_price
     end
 
     def copy_tax_category
-      if variant
-        self.tax_category = variant.tax_category
-      end
+      self.tax_category = variant.try!(:tax_category) || TaxCategory.default
```
Comment (gmacdougall) [inline core/app/models/spree/line_item.rb]: The other option would be to override `.tax_category` on `Spree::Variant` and return `TaxCategory.default` in that case.

Comment (athal7) [inline core/app/models/spree/line_item.rb]: yes, i think I like that better. do you have a preference?

Comment (gmacdougall) [inline core/app/models/spree/line_item.rb]: Not particularly. The nice thing about it being in Variant is that everything else that would use tax categories would now have the appropriate tax category.

Comment (athal7) [inline core/app/models/spree/line_item.rb]: @gmacdougall it turns out that this already happens at the product level: https://github.com/bonobos/spree/blob/2-2-dev/core/app/models/spree/product.rb#L103.

closing

**Pull request closed without merging**